### DISCUSSION TITLE
Virtual Threads / Loom / @RunOnVirtualThread

### DISCUSSION THREAD

**Discussion by bcluap** [quarkusio/quarkus#29211] [Q&A]
I've been doing some research and analysis of various scenarios on Quarkus 2.14 with JDK 19 and virtual threads. I've come across the fact that a RestEasy Reactive endpoint with @RunOnVirtualThread uses a virtual thread, whereas if it also has @Transactional then a platform thread is used. Is this intentional due to the assumption there will be pinning or a bug? I don't see this mentioned anywhere.

Then a second question - I assume that Netty (and VertX) are still using normal platform threads for the event loops and the RestEasy Reactive Loom support is more around whether the executor pool for blocking calls uses virtual or platform threads? Is that right?

Thanks

**Comment by sysid** [comment]
May I add to the question: Is it possible to have the Virtual Thread logged in logging statements? 

Currently log output shows empty `()` where the thread is supposed to be for virtual threads:
```
2023-03-02 08:14:00,374 INFO  [com.xxx.Main] (Quarkus Main Thread) Do startup logic here
2023-03-02 08:14:00,381 DEBUG [com.xxx.bou.SimulationResource] () xxx
```
